auth-test/src/main/kotlin/com/hedvig/android/auth/FakeAuthRepository.kt

```diff
@@ -0,0 +1,54 @@
+package com.hedvig.android.auth
+
+import app.cash.turbine.Turbine
+import com.hedvig.authlib.AuthAttemptResult
+import com.hedvig.authlib.AuthRepository
+import com.hedvig.authlib.AuthTokenResult
+import com.hedvig.authlib.Grant
+import com.hedvig.authlib.LoginMethod
+import com.hedvig.authlib.LoginStatusResult
+import com.hedvig.authlib.ResendOtpResult
+import com.hedvig.authlib.RevokeResult
+import com.hedvig.authlib.StatusUrl
+import com.hedvig.authlib.SubmitOtpResult
+import kotlinx.coroutines.flow.Flow
+
+class FakeAuthRepository : AuthRepository {
+
+  val resendOtpResponse = Turbine<ResendOtpResult>()
+  val submitOtpResponse = Turbine<SubmitOtpResult>()
+  val exchangeResponse = Turbine<AuthTokenResult>()
```

Comment:
Learned this approach from https://www.youtube.com/watch?v=uk8kc9jgzbM&t=1610s mostly.

So this is what enables what I wrote above in a previous comment regarding giving the AuthRepository the ability to respond when `we` want to.

Turbine<T> is basically Channel<T> with some convenience functions on top of it. It allows in our case to simply do `awaitItem()` on one of these which does this. When the test calls this function, it's suspending so it suspends until there's an item. Initially it's empty, so it just waits there for the response. When we do for example `submitOtpResponse.add(Foo)`, the channel gets an item, then the `awaitItem()` suspending function resolves with the result that we passed in, and it continues on with the test given that response. Gives us full control of when and what this fake auth repository responds and enables the tests in the way I wrote in that other bigger comment.